### PR TITLE
Add mel dset support

### PR DESCRIPTION
**Pull request by Or-Tal**
this pull request adds mel spectrogram support - get mel-spec as input instead of raw nb audio. clean remains the target audio waveform

### CODE REVIEW

File: conf/experiment/demucs_1.yaml [hunk 1]
```diff
@@ -28,4 +28,17 @@ demucs:
...
+  feature_model: 'hubert'
+  state_dict_path: '/cs/labs/adiyoss/shared/pretrained_weights/hubert/hubert_base_ls960.pt'
+  features_factor: 0.01
+
+mel:
+  use_melspec: False
+  sample_rate: 16000
+  n_fft: 512
+  n_mels: 128
+  hop_length: 256
```
Comment (mosem) [inline conf/experiment/demucs_1.yaml:44]: end of line

File: conf/experiment/demucs_w_features.yaml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+  resample: 1
+  growth: 2
+  rescale: 0.1
+  scale_factor: ${experiment.scale_factor}
+
+features_model:
+  include_ft: True
+  feature_model: 'hubert'
+  state_dict_path: '/cs/labs/adiyoss/shared/pretrained_weights/hubert/hubert_base_ls960.pt'
+  features_factor: 0.1
```
Comment (mosem) [inline conf/experiment/demucs_w_features.yaml:37]: end of line

File: conf/experiment/example_file.yaml [hunk 3]
```diff
@@ -12,7 +12,7 @@ batch_size: 16
 scale_factor: 1
 
 # models related
-model: demucs # demucs/demucs_adversarial/demucs_hifi/demucs_hifi_features
+model: demucs # demucs/demucs_adversarial
```
Comment (mosem) [inline conf/experiment/example_file.yaml:15]: thanks :)

**Review by mosem — COMMENTED** [review]
OK, I tried merging the master into this branch but I need to wait until the other branches merge. Then it will be much easier for me to review.

**Review by mosem — APPROVED** [review]
I suggest a small fix in train.py, but other than that looks great and ready to be merged to master.

File: train.py [hunk 4]
```diff
@@ -41,24 +43,26 @@ def run(args):
...
+
     target_training_length = int(args.experiment.segment * args.experiment.sample_rate)
     training_stride = int(args.experiment.stride * args.experiment.sample_rate)
     kwargs = {"matching": args.dset.matching, "sample_rate": args.experiment.sample_rate}
     # Building datasets and loaders
     tr_dataset = NoisyCleanSet(args.dset.train, batch_solver.estimate_output_length, clean_length=target_training_length,
-                               stride=training_stride, pad=args.experiment.pad, scale_factor=args.experiment.scale_factor, is_training=True,
-                               **kwargs)
+                               stride=training_stride, pad=args.experiment.pad, scale_factor=args.experiment.scale_factor,
+                               is_training=True, **kwargs, mel_config=mel_config)
```
[the remark this answers is not in the document]
Comment (Or-Tal) [reply train.py:54]: mel_config is a dataclass object

**Pull request merged**